qutip/core/qobj.py

```diff
@@ -1607,17 +1616,35 @@ def eigenstates(
             new_dims = [self._dims[0], [1]]
         else:
             new_dims = [self._dims[0], [1]*len(self.dims[0])]
-        ekets = np.empty((evecs.shape[1],), dtype=object)
-        ekets[:] = [Qobj(vec, dims=new_dims, copy=False)
-                    for vec in _data.split_columns(evecs, False)]
-        norms = np.array([ket.norm() for ket in ekets])
-        if phase_fix is None:
-            phase = np.array([1] * len(ekets))
-        else:
-            phase = np.array([np.abs(ket[phase_fix, 0]) / ket[phase_fix, 0]
-                              if ket[phase_fix, 0] else 1
-                              for ket in ekets])
-        return evals, ekets / norms * phase
+
+        if output_type == 'kets':
+            ekets = np.empty((evecs.shape[1],), dtype=object)
+            ekets[:] = [Qobj(vec, dims=new_dims, copy=False)
+                        for vec in _data.split_columns(evecs, False)]
+            norms = np.array([ket.norm() for ket in ekets])
+            if phase_fix is None:
+                phase = np.array([1] * len(ekets))
+            else:
+                phase = np.array([np.abs(ket[phase_fix, 0]) / ket[phase_fix, 0]
+                                  if ket[phase_fix, 0] else 1
+                                  for ket in ekets])
+            return evals, ekets / norms * phase
```

Comment:
Having a return here feels unbalanced...
Could you rewrite for a single return.
Otherwise looking good.